src/harness/reference_models/fss_purge/fss_purge.py

```diff
@@ -0,0 +1,332 @@
+  cbsd_height = cbsd['registrationRequest']['installationParam']['height']
+  cbsd_height_type = cbsd['registrationRequest']['installationParam']['heightType']
+  cbsd_indoor_dep = cbsd['registrationRequest']['installationParam']['indoorDeployment']
+  if cbsd_height_type == 'AMSL':
+      cbsd_height_agl = convertAmslToAgl(cbsd_latitude, cbsd_longitude, cbsd_height)
+  else:
+      cbsd_height_agl = cbsd_height
+
+  # Invoking ITM propagation model
+  db_loss, incidence_angle, _ = wf_itm.CalcItmPropagationLoss(
+                                         cbsd_latitude,
+                                         cbsd_longitude,
+                                         cbsd_height_agl,
+                                         fss.latitude,
+                                         fss.longitude,
+                                         fss.height_agl,
+                                         cbsd_indoor_dep,
+                                         reliability=-1,
+                                         freq_mhz=FREQ_PROP_MODEL)
+
+  return db_loss, incidence_angle
+
+def convertAmslToAgl(latitude, longitude, height_amsl):
+  """ Convert AMSL height to AGL height."""
+
+  altitude_cbsd = wf_itm.terrainDriver.GetTerrainElevation(latitude, longitude)
+  return height_amsl - altitude_cbsd
+
+def getAntennaGainFssReceiver(fss_entity, incidence_angle):
+  """Get antenna gain of FSS receiver determined using FSS antenna reference model.
+
+  Args:
+    cbsd: A CbsdData object
+    incidence_angle: Incidence angle as identified by ITM model.
+  Returns:
+    Antenna gain of FSS in dBi.
+  """
+
+  #Invoke FSS antenna reference model.
+  fss_ant_gain = antenna.GetFssAntennaGains(
+                              incidence_angle.hor_rx, incidence_angle.ver_rx,
+                              fss_entity.pointing_azimuth,
+                              fss_entity.pointing_elevation,
+                              fss_entity.max_gain_dbi,
+                              fss_entity.weight_1,
+                              fss_entity.weight_2)
+
+
+  return fss_ant_gain
+
+def fssPurgeModel(sas_uut_fad, sas_test_harness_fads, fss_record, fss_oobe_margin):
+  """Entry point function to execute FSS purge list model.
+
+  Performs FSS purge model as described in R2-SGN-29 on the FAD CBSD records of SAS UUT
+  and SAS test harnesses for a given FSS entity. The grants are removed from the CBSD
+  records of the FAD objects.
+
+  Args:
+    sas_uut_fad: A FullActivityDump object containing the FAD records of SAS UUT.
+    sas_test_harness_fads: A list of FullActivityDump objects containing the FAD records 
+      from SAS test harnesses.
+    fss_record: A single FSS record dictionary.
+    fss_oobe_margin: OOBE Margin value to be considered for FSS purge list calculation.
+  """
+
+  # If the FSS is of TT&C type then perform the FSS purge model for the FSS.
+  if fss_record['deploymentParam'][0]['ttc']:

```

Comment:
This parameter is not currently part of an FSS record.

Reply:
FSS record has been updated with this field   as part of this PR.